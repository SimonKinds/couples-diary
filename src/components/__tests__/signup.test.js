// @flow
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Signup from '../signup.js';

describe('<Signup />', () => {
  it('matches its snapshot', () => {
    const tree = renderer
      .create(<Signup />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has an email input', () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper.find(`input[name='email']`).length).toBe(1);
  });

  it('has a submit button', () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper.find(`buttom[name='submit']`).length).toBe(1);
  });
});
