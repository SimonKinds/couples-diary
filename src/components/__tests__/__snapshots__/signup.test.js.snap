// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> matches its snapshot 1`] = `
<form
  className="signup"
>
  Email:
  <input
    name="email"
    type="text"
  />
  <buttom
    name="submit"
    type="submit"
    value="Submit"
  />
</form>
`;
